force-app/main/default/classes/BDI_ContactService.cls

```diff
@@ -1168,4 +1289,117 @@ public with sharing class BDI_ContactService {
         set;
     }
 
+    /*******************************************************************************************************
+    * @description updates DI records with salesforce ids if it finds a match using duplicate management rules
+    */
+    private void performDuplicateManagement() {
```

Comment:
This method can be further broken down into smaller methods to help with readability here. The population of the list of DIs for `Contact1` and `Contact2` can happen later in processing since we only want to create those lists if we found any duplicates

Reply:
Made these changes.